crates/vhost-user-backend/src/vring.rs

```diff
@@ -192,6 +196,16 @@ impl<M: GuestAddressSpace> VringState<M> {
         self.queue.set_next_avail(base);
     }
 
+    /// Set the queue next used head index to the value from the used ring.
+    fn fetch_queue_next_used_from_guest(&mut self) -> Result<(), VirtQueError> {
```

Comment:
I'd prefer a method called `set_queue_next_used()` where you would provide the expected value.
```rust
fn set_queue_next_used(&mut self, idx: u16) {
    self.queue.set_next_used(idx);
}
```
And I'd expect another one for getting the used_idx.

Reply:
Just a clarification, the new function to get the used idx in vring, which index should return the used idx from `self.queue` or from the guest?, I mean should it call:
1- `self.queue.next_used()`
or 
2- `self.queue.used_idx(...)`

Reply:
Well based on your current code, I'd simply define the following:
```rust
/// Get queue used index.
fn queue_used_idx(&self) -> u16 {
    self.queue.used_idx(...)
}
```

Reply:
Done